subprojects/logging/src/main/java/org/gradle/internal/logging/progress/ProgressLogger.java

```diff
@@ -110,7 +113,16 @@
      *
      * @param status The new status message. Can be null or empty.
      */
-    void progress(String status);
+    void progress(@Nullable String status);
+
+    /**
+     * Logs some progress, indicated by a new status with the given build health.
+     *
+     * @param status The new status message. Can be null or empty.
+     * @param progressIndicator A printable string that indicates the progress (e.g. [====------])
+     * @param buildHealth The build health.
+     */
+    void progress(@Nullable String status, @Nullable String progressIndicator, BuildHealth buildHealth);
```

Comment:
ProgressEvent is still extended now but with the general progress information (see comment above).